src/main/scala/com/ing/wbaa/rokku/proxy/api/directive/ProxyDirectives.scala

```diff
@@ -32,55 +35,76 @@ object ProxyDirectives extends LazyLogging {
     if (httpHeader.is(AUTHORIZATION_HTTP_HEADER_NAME)) {
       val signerType = httpHeader.value().split(" ").headOption
       logger.debug("Signertype used: {}", signerType)
+      getAccessKeyFromAuthString(httpHeader.value(), signerType)
+    } else None
 
-      signerType match {
-        case Some("AWS4-HMAC-SHA256") =>
-          val credential =
-            """\S+ Credential=(\S+), """.r
-              .findFirstMatchIn(httpHeader.value())
-              .map(_ group 1)
+  private[this] def getAccessKeyFromAuthString(authString: String, signerType: Option[String]): Option[AwsAccessKey] = {
+    signerType match {
+      case Some("AWS4-HMAC-SHA256") =>
+        val credential =
+          """\S+ Credential=(\S+), """.r
```

Comment:
Could we simplify regex to just `Credential=(+S)`? According to [documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html) it will be unique enough for this purpose and also will work if not in correct order